terratorch/tasks/segmentation_tasks.py

```diff
@@ -98,22 +99,47 @@ def __init__(
         self.tiled_inference_parameters = tiled_inference_parameters
         self.aux_loss = aux_loss
         self.aux_heads = aux_heads
-        self.model_factory = MODEL_FACTORY_REGISTRY.build(model_factory)
+        self._model_module = None
+
+        if model_factory:  
+            self.model_factory = MODEL_FACTORY_REGISTRY.build(model_factory)
+            self.model_builder = self._build
+        else:
+            self.model_builder = self._bypass_build
+
+        self._model_module = None 
+
         super().__init__()
+
+        self._model_module = model
+        self.model = model
+        
         self.train_loss_handler = LossHandler(self.train_metrics.prefix)
         self.test_loss_handler = LossHandler(self.test_metrics.prefix)
         self.val_loss_handler = LossHandler(self.val_metrics.prefix)
         self.monitor = f"{self.val_metrics.prefix}loss"
         self.plot_on_val = int(plot_on_val)
 
-    # overwrite early stopping
+    @property
+    def model_module(self):
+        return self._model_module
+
+       # overwrite early stopping
     def configure_callbacks(self) -> list[Callback]:
         return []
 
-    def configure_models(self) -> None:
-        self.model: Model = self.model_factory.build_model(
+    def _bypass_build(self):
+        return self.model_module
+
+    def _build(self):
+
+        return self.model_factory.build_model(
             "segmentation", aux_decoders=self.aux_heads, **self.hparams["model_args"]
         )
+
+    def configure_models(self) -> None:
+        self.model: Model = self.model_builder()
+
```

Comment:
So are you sure this works?

Case 1:
You pass a modelfactory, then self.model_builder = self._build and the model is build from the factory

Case 2:
You don't pass a model factory but a model
self.model_builder = self._bypass_build and the one you get passed in from constructor is used

I think the code is correct but hard to understand, I also might wanna add some argument checking, e.g. if model factory is None a model need to be passed

Reply:
When no model_factory is passed, it used the object model, so the the method self._bypass_build just return this already instantiated object to be used during the process.